src/main/java/org/junitpioneer/jupiter/report/PioneerReport.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.report;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlType;
+
+/**
+ * <p>Java class for anonymous complex type.
+ *
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ *
+ * <pre>
+ * &lt;complexType&gt;
+ *   &lt;complexContent&gt;
+ *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType"&gt;
+ *       &lt;sequence&gt;
+ *         &lt;element ref="{}properties" minOccurs="0"/&gt;
+ *         &lt;element ref="{}testcontainer" maxOccurs="unbounded" minOccurs="0"/&gt;
+ *         &lt;element ref="{}testcase" maxOccurs="unbounded" minOccurs="0"/&gt;
+ *       &lt;/sequence&gt;
+ *       &lt;attribute name="name" type="{http://www.w3.org/2001/XMLSchema}string" /&gt;
+ *     &lt;/restriction&gt;
+ *   &lt;/complexContent&gt;
+ * &lt;/complexType&gt;
+ * </pre>
+ *
+ *
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "", propOrder = { "properties", "testcontainer", "testcase" })
+@XmlRootElement(name = "pioneerReport")
+public class PioneerReport {
+
+	protected Properties properties;
+	protected List<Testcontainer> testcontainer;
+	protected List<Testcase> testcase;
+	@XmlAttribute(name = "name")
+	protected String name;
+
+	public PioneerReport() {
+		this.testcontainer = new ArrayList<Testcontainer>();
```

Comment:
Sonar says this can be replaced with the diamond operator `<>` and I have to agree.

Reply:
As the report files are generated I suggest to exclude all of them.

Reply:
I changed the diamond operators, but I won't change the "code duplication", because they are no duplication.